extensions/ql-vscode/src/model-editor/method-modeling/method-modeling-view-provider.ts

```diff
@@ -79,4 +101,45 @@ export class MethodModelingViewProvider
         break;
     }
   }
+
+  private registerToModelingStoreEvents(): void {
+    this.modelingStore.onModeledMethodsChanged(async (e) => {
+      if (this.webviewView && e.isActiveDb) {
+        const modeledMethod = e.modeledMethods[this.method?.signature ?? ""];
+        if (modeledMethod) {
+          await this.webviewView.webview.postMessage({
+            t: "setModeledMethod",
+            method: modeledMethod,
+          });
+        }
+      }
+    });
+
+    this.modelingStore.onModifiedMethodsChanged(async (e) => {
+      if (
+        this.webviewView &&
+        e.isActiveDb &&
+        this.method &&
+        this.method.signature
```

Comment:
This would be a lot shorter since we could move the complete if-condition to 1 line:

```suggestion
        this.method?.signature
```

However, I'm wondering why we're checking whether the method signature is non-empty. Do we have cases where we would have a method without a signature?

Reply:
No that shouldn't be possible, you're right! I'll remove the signature check.